File: leetcode/java/0001_two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (hMap.containsKey(target - nums[i])) {
                return new int[] {hMap.get(target - nums[i]), i};
            } else {
                hMap.put(nums[i], i);
            }
        }

        return new int[] {};
    }
}

File: leetcode/cpp/0001_two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hashMap;
        vector<int> ret;
        
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            
            if (hashMap.find(diff) != hashMap.end()) {
                auto secondIndex = (hashMap.find(diff))->second;
                ret.push_back(secondIndex);
                
                ret.push_back(i);
            }
            else {
                hashMap.insert(make_pair(nums[i], i));
            }
        }
        
        return ret;
    }
};